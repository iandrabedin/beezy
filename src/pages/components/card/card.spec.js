import React from "react";
import { cleanup, render } from "@testing-library/react";
import Card from "./card";

afterEach(cleanup);

const props = {
  handleSelectedMovie: jest.fn(),
  poster: "",
  title: "movie title",
  date: "1995",
  overview: "overview",
  rating: 88,
  id: 270,
};

describe("Card", () => {
  it("should display all data", () => {
    const { getByText } = render(<Card {...props} />);
    expect(getByText("movie title")).toBeDefined();
    expect(getByText("1995")).toBeDefined();
    expect(getByText("88")).toBeDefined();
  });

  it("should have the more button", () => {
    const { getByText } = render(<Card {...props} />);
    expect(getByText("More info")).toBeDefined();
  });

  it("should render as expected", () => {
    const { asFragment } = render(<Card {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
