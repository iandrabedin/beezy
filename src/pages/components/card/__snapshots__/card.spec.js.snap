// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render as expected 1`] = `
<DocumentFragment>
  <div
    class="col-sm-12 col-lg-6"
  >
    <div
      class="card"
    >
      <div
        class="row"
      >
        <div
          class="col-sm-12 col-lg-4"
        >
          <div
            class="image"
          />
        </div>
        <div
          class="col-sm-12 col-lg-6 offset-lg-1"
        >
          <div
            class="info"
          >
            <div
              class="row"
            >
              <div
                class="col-sm-2 col-lg-2"
              >
                <div
                  class="rating"
                >
                  88
                </div>
              </div>
              <div
                class="col-sm-10 col-lg-10"
              >
                <h3
                  class="title"
                >
                  movie title
                </h3>
                <p
                  class="subtitle"
                >
                  1995
                </p>
              </div>
            </div>
            <div
              class="overview"
            >
              <p>
                overview
              </p>
            </div>
            <button
              class="button button-text"
            >
              More info
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
